batchservice/src/main/java/nu/yona/server/batch/jobs/ActivityAggregationBatchJob.java

```diff
@@ -0,0 +1,176 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+import nu.yona.server.analysis.entities.DayActivity;
+import nu.yona.server.analysis.entities.WeekActivity;
+import nu.yona.server.exceptions.YonaException;
+import nu.yona.server.util.TimeUtil;
+
+@Component
+public class ActivityAggregationBatchJob
+{
+	private static final Logger logger = LoggerFactory.getLogger(ActivityAggregationBatchJob.class);
+
+	private static final int DAY_ACTIVITY_CHUNK_SIZE = 10;
+	private static final int WEEK_ACTIVITY_CHUNK_SIZE = 10;
+	// once we have users in multiple zones, the batch job should become sensitive to user time zone
+	private static final ZoneId DEFAULT_TIME_ZONE = ZoneId.of("Europe/Amsterdam");
+
+	@Autowired
+	private JobBuilderFactory jobBuilderFactory;
+
+	@Autowired
+	private StepBuilderFactory stepBuilderFactory;
+
+	@PersistenceContext
+	private EntityManager entityManager;
+
+	@Autowired
+	@Qualifier("activityAggregationJobDayActivityReader")
+	private JpaPagingItemReader<DayActivity> dayActivityReader;
+
+	@Autowired
+	@Qualifier("activityAggregationJobWeekActivityReader")
+	private JpaPagingItemReader<WeekActivity> weekActivityReader;
+
+	@Bean("activityAggregationJob")
+	public Job activityAggregationBatchJob()
+	{
+		return jobBuilderFactory.get("activityAggregationBatchJob").incrementer(new RunIdIncrementer())
+				.flow(aggregateDayActivities()).next(aggregateWeekActivities()).end().build();
+	}
+
+	private Step aggregateDayActivities()
+	{
+		return stepBuilderFactory.get("aggregateDayActivities").<DayActivity, DayActivity> chunk(DAY_ACTIVITY_CHUNK_SIZE)
+				.reader(dayActivityReader).processor(dayActivityProcessor()).writer(dayActivityWriter()).build();
+	}
+
+	private Step aggregateWeekActivities()
+	{
+		return stepBuilderFactory.get("aggregateWeekActivities").<WeekActivity, WeekActivity> chunk(WEEK_ACTIVITY_CHUNK_SIZE)
+				.reader(weekActivityReader).processor(weekActivityProcessor()).writer(weekActivityWriter()).build();
+	}
+
+	@Bean(name = "activityAggregationJobDayActivityReader", destroyMethod = "")
+	@StepScope
+	public JpaPagingItemReader<DayActivity> dayActivityReader()
+	{
+		try
+		{
+			String jpqlQuery = "SELECT d FROM DayActivity d WHERE d.startDate <= :yesterday AND d.aggregatesComputed = false";
+
+			JpaPagingItemReader<DayActivity> reader = new JpaPagingItemReader<>();
+			reader.setQueryString(jpqlQuery);
+			reader.setParameterValues(
+					Collections.singletonMap("yesterday", ZonedDateTime.now(DEFAULT_TIME_ZONE).minusDays(1).toLocalDate()));
+			reader.setEntityManagerFactory(entityManager.getEntityManagerFactory());
+			reader.setPageSize(DAY_ACTIVITY_CHUNK_SIZE);
+			reader.afterPropertiesSet();
+			reader.setSaveState(true);
+
+			return reader;
+		}
+		catch (Exception e)
```

Comment:
Actually, the `JpaPagingItemReader.afterPropertiesSet` has `throws Exception`, so it cannot be made more specific.